File: demo/src/main/java/com/example/demo/entity/Comment.java
package com.example.demo.entity;

import java.time.OffsetDateTime;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;

import lombok.Data;

/**
 * 评论实体（文档批注与回复）
 * 对应数据库表：comments
 */
@Data
@TableName("comments")
public class Comment {
    
    /**
     * 评论ID（主键）
     */
    @TableId(type = IdType.AUTO)
    private Long id;
    
    /**
     * 文档ID
     */
    @TableField("document_id")
    private Long documentId;
    
    /**
     * 评论人ID
     */
    @TableField("user_id")
    private Long userId;
    
    /**
     * 评论内容
     */
    @TableField("content")
    private String content;
    
    /**
     * 回复目标评论ID
     */
    @TableField("reply_to_comment_id")
    private Long replyToCommentId;
    
    /**
     * 选中范围信息（JSON格式）
     */
    @TableField("range_info")
    private String rangeInfo;
    
    /**
     * 评论状态：open-未解决，resolved-已解决
     */
    @TableField("status")
    private String status;
    
    /**
     * 创建时间
     */
    @TableField("created_at")
    private OffsetDateTime createdAt;
    
    /**
     * 更新时间
     */
    @TableField("updated_at")
    private OffsetDateTime updatedAt;
}

File: frontend/lib/api/comment.ts
import axiosInstance from '../axios';

export interface Comment {
  id: number;
  documentId: number;
  userId: number;
  parentId?: number;
  content: string;
  rangeInfo?: string;
  status: 'OPEN' | 'RESOLVED';
  createdAt: string;
  updatedAt: string;
}

export interface CommentVO extends Comment {
  username: string;
  userAvatar?: string;
  replies: CommentVO[];
}

export interface CreateCommentRequest {
  documentId: number;
  parentId?: number;
  content: string;
  rangeInfo?: string;
}

/**
 * 创建评论
 */
export const createComment = async (data: CreateCommentRequest): Promise<CommentVO> => {
  const response = await axiosInstance.post('/api/comments', data);
  return response.data.data;
};

/**
 * 获取文档评论列表
 */
export const getCommentsByDocument = async (documentId: number): Promise<CommentVO[]> => {
  const response = await axiosInstance.get(`/api/comments/document/${documentId}`);
  return response.data.data;
};

/**
 * 标记评论为已解决
 */
export const resolveComment = async (id: number): Promise<void> => {
  await axiosInstance.put(`/api/comments/${id}/resolve`);
};

/**
 * 删除评论
 */
export const deleteComment = async (id: number): Promise<void> => {
  await axiosInstance.delete(`/api/comments/${id}`);
};
